File: PageObjects/ProductCatalogPage.cs
using Microsoft.Playwright;

namespace PerfectDraftTests.PageObjects
{
    public class ProductCatalogPage : BasePage
    {
        public ProductCatalogPage(IPage page) : base(page)
        {
        }

        public async Task NavigateToKegsSection()
        {
            // Try multiple selectors to find the kegs/beer section
            var kegSelectors = new[]
            {
                "text=Kegs",
                "text=Beer Kegs",
                "a[href*='kegs']",
                "a[href*='beer']",
                "[data-category='kegs']",
                ".navigation a:has-text('Kegs')",
                ".menu-item:has-text('Kegs')"
            };

            bool navigationSuccessful = false;

            foreach (var selector in kegSelectors)
            {
                try
                {
                    var element = await FindElementAsync(selector);
                    if (await element.IsVisibleAsync())
                    {
                        await element.ClickAsync();
                        await WaitForPageLoadAsync();
                        navigationSuccessful = true;
                        Console.WriteLine(selector);
                        break;
                    }
                }
                catch
                {
                    continue;
                }
            }

            if (!navigationSuccessful)
            {
                // Fallback: try to navigate via URL
                var currentUrl = await GetCurrentUrlAsync();
                var baseUrl = new Uri(currentUrl).GetLeftPart(UriPartial.Authority);
                await NavigateToAsync($"{baseUrl}/kegs");
            }
        }

        public async Task<bool> IsKegListVisible()
        {
            var kegListSelectors = new[]
            {
                ".ais-InfiniteHits"
            };

            foreach (var selector in kegListSelectors)
            {
                try
                {
                    // Use Playwright's built-in waiting with 30 second timeout
                    await WaitForElementToBeVisibleAsync(selector, 30000);
                    return true;
                }
                catch (TimeoutException)
                {
                    // Continue to next selector if this one times out
                    continue;
                }
                catch
                {
                    continue;
                }
            }

            return false;
        }

        public async Task<int> GetKegCount()
        {
            var kegSelectors = new[]
            {
                ".result-wrapper",
            };

            foreach (var selector in kegSelectors)
            {
                try
                {
                    // Wait for at least one keg to be present using Playwright's built-in waiting
                    var locator = Page.Locator(selector);
                    await locator.First.WaitForAsync(new LocatorWaitForOptions
                    {
                        State = WaitForSelectorState.Attached,
                        Timeout = 30000
                    });

                    var count = await locator.CountAsync();
                    if (count > 0)
                    {
                        return count;
                    }
                }
                catch (TimeoutException)
                {
                    // Continue to next selector if this one times out
                    continue;
                }
                catch
                {
                    continue;
                }
            }

            return 0;
        }

        public async Task<bool> DoKegsDisplayRequiredInformation()
        {
            var kegItems = await GetKegElements();

            if (kegItems.Count == 0)
            {
                return false;
            }

            // Check first few kegs for required information
            var kegsToCheck = Math.Min(3, kegItems.Count);

            for (int i = 0; i < kegsToCheck; i++)
            {
                var keg = kegItems[i];

                // Check for product image
                var hasImage = await HasProductImage(keg);

                // Check for beer name/title
                var hasName = await HasProductName(keg);

                // Check for price
                var hasPrice = await HasPriceInfo(keg);

                // Check for ABV (alcohol by volume)
                var hasABV = await HasABVInfo(keg);

                // If at least most information is present, consider it valid
                var infoCount = (hasImage ? 1 : 0) + (hasName ? 1 : 0) +
                               (hasPrice ? 1 : 0) + (hasABV ? 1 : 0);

                if (infoCount < 4) // At least 4 out of 6 pieces of info should be present
                {
                    return false;
                }
            }

            return true;
        }

        private async Task<IReadOnlyList<ILocator>> GetKegElements()
        {
            var elements = Page.Locator(".result-wrapper");

            if (await elements.CountAsync() > 0)
            {
                return await elements.AllAsync();
            }

            return new List<ILocator>();
        }

        private async Task<bool> HasProductImage(ILocator kegElement)
        {
            var image = kegElement.Locator("img[itemprop=image]");
            return await image.CountAsync() > 0 && await image.First.IsVisibleAsync();
        }

        private async Task<bool> HasProductName(ILocator kegElement)
        {
            var nameSelectors = new[]
            {
                ".result-title"
            };

            return await HasTextContent(kegElement, nameSelectors);
        }

        private async Task<bool> HasPriceInfo(ILocator kegElement)
        {
            var priceSelectors = new[]
            {
                ".price",
            };

            return await HasTextContent(kegElement, priceSelectors, "£");
        }

        private async Task<bool> HasABVInfo(ILocator kegElement)
        {
            var abvSelectors = new[]
            {
                ".attr-label__abv",
            };

            return await HasTextContent(kegElement, abvSelectors, "%");
        }

        private async Task<bool> HasTextContent(ILocator kegElement, string[] selectors, string? mustContain = null)
        {
            foreach (var selector in selectors)
            {
                try
                {
                    var element = kegElement.Locator(selector);
                    if (await element.CountAsync() > 0)
                    {
                        var text = await element.First.TextContentAsync();
                        if (!string.IsNullOrWhiteSpace(text))
                        {
                            if (mustContain == null || text.Contains(mustContain))
                            {
                                return true;
                            }
                        }
                    }
                }
                catch
                {
                    continue;
                }
            }
            return false;
        }

        public async Task<bool> AreFiltersAvailable()
        {
            var filterSelectors = new[]
            {
                ".filter",
                ".filters",
                ".product-filters",
                "[data-testid='filters']",
                ".filter-container",
                "select[name*='filter']",
                ".filter-dropdown"
            };

            foreach (var selector in filterSelectors)
            {
                try
                {
                    if (await IsElementVisibleAsync(selector))
                    {
                        return true;
                    }
                }
                catch
                {
                    continue;
                }
            }

            // Also check for filter-related text
            var filterTexts = new[] { "Filter", "Filter by", "Beer Type", "Brand" };
            foreach (var text in filterTexts)
            {
                try
                {
                    var element = Page.GetByText(text, new() { Exact = false });
                    if (await element.CountAsync() > 0 && await element.First.IsVisibleAsync())
                    {
                        return true;
                    }
                }
                catch
                {
                    continue;
                }
            }

            return false;
        }

        public async Task<bool> AreSortOptionsAvailable()
        {
            var sortSelectors = new[]
            {
                ".sort",
                ".sort-by",
                ".product-sort",
                "[data-testid='sort']",
                "select[name*='sort']",
                ".sort-dropdown"
            };

            foreach (var selector in sortSelectors)
            {
                try
                {
                    if (await IsElementVisibleAsync(selector))
                    {
                        return true;
                    }
                }
                catch
                {
                    continue;
                }
            }

            // Also check for sort-related text
            var sortTexts = new[] { "Sort", "Sort by", "Price", "Popularity", "Name" };
            foreach (var text in sortTexts)
            {
                try
                {
                    var element = Page.GetByText(text, new() { Exact = false });
                    if (await element.CountAsync() > 0 && await element.First.IsVisibleAsync())
                    {
                        return true;
                    }
                }
                catch
                {
                    continue;
                }
            }

            return false;
        }

        public async Task<bool> IsMachineTypeVisible(string machineType)
        {
            if (machineType == "PerfectDraft")
            {
                // For standard PerfectDraft, find text that contains "PerfectDraft" 
                // but does NOT contain "Pro" or "Black"
                var allPerfectDraftElements = await Page.GetByText("PerfectDraft").AllAsync();
                
                foreach (var element in allPerfectDraftElements)
                {
                    if (await element.IsVisibleAsync())
                    {
                        var text = await element.TextContentAsync();
                        if (text != null && 
                            text.Contains("PerfectDraft") && 
                            !text.Contains("Pro") && 
                            !text.Contains("Black"))
                        {
                            return true;
                        }
                    }
                }
                return false;
            }
            else
            {
                // For other machine types (like "PerfectDraft Pro"), use contains match
                try
                {
                    var element = Page.GetByText(machineType, new() { Exact = false });
                    return await element.CountAsync() > 0 && await element.First.IsVisibleAsync();
                }
                catch
                {
                    return false;
                }
            }
        }

        public async Task<bool> IsComparisonLinkVisible()
        {
            try
            {
                var comparisonLink = Page.Locator("#dy-banner-104346742");
                return await comparisonLink.CountAsync() > 0 && await comparisonLink.First.IsVisibleAsync();
            }
            catch
            {
                return false;
            }
        }

        public async Task ClickOnFirstMachine()
        {
            try
            {
                // Click on the actual link element instead of the content div
                var firstMachineLink = Page.Locator("a.result.product__link").First;
                await firstMachineLink.ClickAsync();
                await WaitForPageLoadAsync();
            }
            catch (Exception ex)
            {
                throw new InvalidOperationException($"Failed to click on first machine: {ex.Message}", ex);
            }
        }

        public async Task<bool> AreSpecificationsVisible()
        {
            try
            {
                var specsContainer = Page.Locator(".specs");
                return await specsContainer.CountAsync() > 0 && await specsContainer.First.IsVisibleAsync();
            }
            catch
            {
                return false;
            }
        }

        public async Task<bool> IsKegSizeSpecificationVisible()
        {
            try
            {
                var kegSizeElement = Page.Locator("[data-code=\"bottle_size\"]");
                return await kegSizeElement.CountAsync() > 0 && await kegSizeElement.First.IsVisibleAsync();
            }
            catch
            {
                return false;
            }
        }

        public async Task ClickOnProductByName(string productName)
        {
            try
            {
                // First wait for products to load
                await WaitForElementToBeVisibleAsync(".result-wrapper", 10000);

                // Simple approach - just click on first Stella Artois link found
                if (productName.Contains("Stella"))
                {
                    // Find link containing "stella-artois" in href
                    var productLink = Page.Locator("a.result.product__link[href*='stella-artois']").First;
                    await productLink.ClickAsync();
                }
                else
                {
                    // For other products, try to find by partial name match in href
                    var searchName = productName.Split(' ')[0].ToLower();
                    var productLink = Page.Locator($"a.result.product__link[href*='{searchName}']").First;
                    await productLink.ClickAsync();
                }

                await WaitForPageLoadAsync();
            }
            catch (Exception ex)
            {
                throw new InvalidOperationException($"Failed to click on product '{productName}': {ex.Message}", ex);
            }
        }

        public async Task AddProductToCart(string productName)
        {
            try
            {
                // Wait for products to load
                await WaitForElementToBeVisibleAsync(".result-wrapper", 10000);

                // Multiple selectors for add to cart button
                var addToCartSelectors = new[]
                {
                    ".action.tocart.primary",
                    "button[title*='Add to']",
                    "button:has-text('Add to cart')",
                    "button:has-text('Add to basket')",
                    ".btn-cart",
                    "[data-action='add-to-cart']"
                };

                // If product name is specified, find the specific product container first
                if (!string.IsNullOrEmpty(productName))
                {
                    ILocator productContainer;
                    
                    if (productName.Contains("Stella"))
                    {
                        productContainer = Page.Locator(".result-wrapper").Filter(new() { HasText = "Stella" });
                    }
                    else
                    {
                        var searchName = productName.Split(' ')[0];
                        productContainer = Page.Locator(".result-wrapper").Filter(new() { HasText = searchName });
                    }

                    // Try to find add to cart button within the product container
                    foreach (var selector in addToCartSelectors)
                    {
                        try
                        {
                            var button = productContainer.Locator(selector).First;
                            if (await button.IsVisibleAsync())
                            {
                                await button.ClickAsync();
                                return;
                            }
                        }
                        catch
                        {
                            continue;
                        }
                    }
                }

                // Fallback: try to find any add to cart button on the page
                foreach (var selector in addToCartSelectors)
                {
                    try
                    {
                        var button = Page.Locator(selector).First;
                        if (await button.IsVisibleAsync())
                        {
                            await button.ClickAsync();
                            return;
                        }
                    }
                    catch
                    {
                        continue;
                    }
                }

                // If still not found, it might be on product detail page
                // Navigate to product detail page first
                await ClickOnProductByName(productName);
                
                // Now try to add from product detail page
                var pdpAddButton = Page.Locator("#product-addtocart-button, .action.tocart.primary").First;
                if (await pdpAddButton.IsVisibleAsync())
                {
                    await pdpAddButton.ClickAsync();
                }
                else
                {
                    throw new InvalidOperationException("Could not find add to cart button");
                }
            }
            catch (Exception ex)
            {
                throw new InvalidOperationException($"Failed to add product '{productName}' to cart: {ex.Message}", ex);
            }
        }
    }
}

File: src/pages/product-catalog.page.js
// src/pages/product-catalog.page.js
import BasePage from './base.page.js';

/**
 * Product Catalog Page Object
 * Represents the product listing/catalog page
 */
class ProductCatalogPage extends BasePage {
  constructor(page) {
    super(page);

    // Specific selectors based on actual website inspection
    this.kegNavigationSelector = 'a.main-nav__link:has-text("Beer Kegs")';
    this.kegListSelector = '.result-wrapper';
    this.filterSelector = '.filter';
    this.sortSelector = '.sort';

    // Product information selectors
    this.productInfoSelectors = {
      image: 'img[itemprop="image"]',
      name: '.result-title',
      price: '.price',
      abv: '.attr-label__abv'
    };
  }

  /**
   * Navigate to kegs section
   */
  async navigateToKegsSection() {
    try {
      // First method: Click the navigation link
      const element = this.findElement(this.kegNavigationSelector);
      if (await element.isVisible({ timeout: 5000 })) {
        await element.click();
        await this.waitForPageLoad();
        return;
      }
    } catch (error) {
      console.log('Navigation link not found, navigating directly via URL');
    }
    
    // Direct navigation as primary method if link isn't visible
    await this.navigateTo('https://www.perfectdraft.com/en-gb/perfect-draft-range/perfect-draft-kegs');
    await this.waitForPageLoad();
  }

  /**
   * Check if keg list is visible
   * @returns {Promise<boolean>}
   */
  async isKegListVisible() {
    try {
      // Use Playwright's built-in waiting with 30 second timeout
      await this.waitForElementToBeVisible(this.kegListSelector, 30000);
      return true;
    } catch {
      return false;
    }
  }

  /**
   * Get keg count
   * @returns {Promise<number>}
   */
  async getKegCount() {
    try {
      // Wait for at least one keg to be present
      const locator = this.page.locator(this.kegListSelector);
      await locator.first().waitFor({
        state: 'attached',
        timeout: 30000
      });

      return await locator.count();
    } catch {
      return 0;
    }
  }

  /**
   * Check if kegs display required information
   * @returns {Promise<boolean>}
   */
  async doKegsDisplayRequiredInformation() {
    const kegElements = await this.getKegElements();

    if (kegElements.length === 0) {
      return false;
    }

    // Check first few kegs for required information
    const kegsToCheck = Math.min(3, kegElements.length);

    for (let i = 0; i < kegsToCheck; i++) {
      const keg = kegElements[i];

      // Check for product image
      const hasImage = await this.hasProductImage(keg);

      // Check for beer name/title
      const hasName = await this.hasProductName(keg);

      // Check for price
      const hasPrice = await this.hasPriceInfo(keg);

      // Check for ABV (alcohol by volume)
      const hasABV = await this.hasABVInfo(keg);

      // If at least most information is present, consider it valid
      const infoCount = (hasImage ? 1 : 0) + (hasName ? 1 : 0) +
        (hasPrice ? 1 : 0) + (hasABV ? 1 : 0);

      if (infoCount < 3) { // At least 3 out of 4 pieces of info should be present
        return false;
      }
    }

    return true;
  }

  /**
   * Get keg elements
   * @returns {Promise<Array>}
   */
  async getKegElements() {
    const elements = this.page.locator(this.kegListSelector);
    const count = await elements.count();

    if (count > 0) {
      const elementArray = [];
      for (let i = 0; i < count; i++) {
        elementArray.push(elements.nth(i));
      }
      return elementArray;
    }

    return [];
  }

  /**
   * Check if product has image
   * @param {Object} kegElement - Locator for keg element
   * @returns {Promise<boolean>}
   */
  async hasProductImage(kegElement) {
    const image = kegElement.locator(this.productInfoSelectors.image);
    const count = await image.count();
    return count > 0 && await image.first().isVisible();
  }

  /**
   * Check if product has name
   * @param {Object} kegElement - Locator for keg element
   * @returns {Promise<boolean>}
   */
  async hasProductName(kegElement) {
    return await this.hasTextContent(kegElement, [this.productInfoSelectors.name]);
  }

  /**
   * Check if product has price info
   * @param {Object} kegElement - Locator for keg element
   * @returns {Promise<boolean>}
   */
  async hasPriceInfo(kegElement) {
    return await this.hasTextContent(kegElement, [this.productInfoSelectors.price], '£');
  }

  /**
   * Check if product has ABV info
   * @param {Object} kegElement - Locator for keg element
   * @returns {Promise<boolean>}
   */
  async hasABVInfo(kegElement) {
    return await this.hasTextContent(kegElement, [this.productInfoSelectors.abv], '%');
  }

  /**
   * Check if element has text content
   * @param {Object} kegElement - Locator for keg element
   * @param {Array} selectors - Array of selectors to try
   * @param {string} mustContain - Text that must be contained
   * @returns {Promise<boolean>}
   */
  async hasTextContent(kegElement, selectors, mustContain = null) {
    // Note: selectors is an array for compatibility, but should only have one element
    const selector = selectors[0];
    try {
      const element = kegElement.locator(selector);
      const count = await element.count();
      if (count > 0) {
        const text = await element.first().textContent();
        if (text && text.trim()) {
          if (mustContain === null || text.includes(mustContain)) {
            return true;
          }
        }
      }
    } catch {
      return false;
    }
    return false;
  }

  /**
   * Check if filters are available
   * @returns {Promise<boolean>}
   */
  async areFiltersAvailable() {
    try {
      // Check for filter elements - look for filter text which is more reliable
      const filterElement = this.page.getByText('Filter', { exact: false });
      const count = await filterElement.count();
      return count > 0 && await filterElement.first().isVisible();
    } catch {
      return false;
    }
  }

  /**
   * Check if sort options are available
   * @returns {Promise<boolean>}
   */
  async areSortOptionsAvailable() {
    try {
      // Check for sort elements - look for sort text which is more reliable
      const sortElement = this.page.getByText('Sort', { exact: false });
      const count = await sortElement.count();
      return count > 0 && await sortElement.first().isVisible();
    } catch {
      return false;
    }
  }

  /**
   * Check if machine type is visible
   * @param {string} machineType - Machine type to check
   * @returns {Promise<boolean>}
   */
  async isMachineTypeVisible(machineType) {
    try {
      // Look for product headings that contain the machine type
      const productHeadings = this.page.locator('h3');
      const headingCount = await productHeadings.count();

      for (let i = 0; i < headingCount; i++) {
        const heading = productHeadings.nth(i);
        if (await heading.isVisible()) {
          const text = await heading.textContent();
          if (text) {
            if (machineType === 'PerfectDraft') {
              // For standard PerfectDraft, find headings that contain "PerfectDraft" 
              // but NOT "Pro" or "Black"
              if (text.includes('PerfectDraft') &&
                !text.includes('Pro') &&
                !text.includes('Black')) {
                return true;
              }
            } else if (machineType === 'PerfectDraft Pro') {
              // Look for "PerfectDraft Pro" in headings
              if (text.includes('PerfectDraft Pro')) {
                return true;
              }
            } else if (machineType === 'PerfectDraft Black') {
              // Look for "PerfectDraft Black" in headings
              if (text.includes('PerfectDraft Black')) {
                return true;
              }
            }
          }
        }
      }
      return false;
    } catch (error) {
      console.log(`Error checking machine type visibility: ${error}`);
      return false;
    }
  }

  /**
   * Check if comparison link is visible
   * @returns {Promise<boolean>}
   */
  async isComparisonLinkVisible() {
    try {
      const comparisonLink = this.page.locator('#dy-banner-104346742');
      const count = await comparisonLink.count();
      return count > 0 && await comparisonLink.first().isVisible();
    } catch {
      return false;
    }
  }

  /**
   * Click on first machine
   */
  async clickOnFirstMachine() {
    try {
      // Click on the actual link element instead of the content div
      const firstMachineLink = this.page.locator('a.result.product__link').first();
      await firstMachineLink.click();
      await this.waitForPageLoad();
    } catch (error) {
      throw new Error(`Failed to click on first machine: ${error.message}`);
    }
  }

  /**
   * Check if specifications are visible
   * @returns {Promise<boolean>}
   */
  async areSpecificationsVisible() {
    try {
      const specsContainer = this.page.locator('.specs');
      const count = await specsContainer.count();
      return count > 0 && await specsContainer.first().isVisible();
    } catch {
      return false;
    }
  }

  /**
   * Check if keg size specification is visible
   * @returns {Promise<boolean>}
   */
  async isKegSizeSpecificationVisible() {
    try {
      const kegSizeElement = this.page.locator('[data-code="bottle_size"]');
      const count = await kegSizeElement.count();
      return count > 0 && await kegSizeElement.first().isVisible();
    } catch {
      return false;
    }
  }

  /**
   * Click on product by name
   * @param {string} productName - Product name
   */
  async clickOnProductByName(productName) {
    try {
      // First wait for products to load
      await this.waitForElementToBeVisible('.result-wrapper', 10000);

      // Handle specific product types
      if (productName.includes('Stella')) {
        // Find link containing "stella-artois" in href
        const productLink = this.page.locator('a.result.product__link[href*="stella-artois"]').first();
        await productLink.click();
      } else if (productName.includes('Camden')) {
        // Find link containing "camden" in href
        const productLink = this.page.locator('a.result.product__link[href*="camden"]').first();
        await productLink.click();
      } else {
        // For other products, try to find by partial name match in href
        const searchName = productName.split(' ')[0].toLowerCase();
        const productLink = this.page.locator(`a.result.product__link[href*="${searchName}"]`).first();
        await productLink.click();
      }

      await this.waitForPageLoad();
    } catch (error) {
      throw new Error(`Failed to click on product '${productName}': ${error.message}`);
    }
  }

  /**
   * Add product to cart
   * @param {string} productName - Product name
   */
  async addProductToCart(productName) {
    // Wait for products to load
    await this.waitForElementToBeVisible('.result.product__link', 10000);

    let addToCartButton;

    if (productName.includes('Stella')) {
      // Find Stella Artois add to cart button by data attributes
      addToCartButton = this.page.locator('button[data-name*="Stella Artois"]');
    } else if (productName.includes('Corona')) {
      addToCartButton = this.page.locator('button[data-name*="Corona"]');
    } else if (productName.includes('Camden')) {
      addToCartButton = this.page.locator('button[data-name*="Camden"]');
    } else {
      // For other products, try to find by partial name match
      const searchName = productName.split(' ')[0];
      addToCartButton = this.page.locator(`button[data-name*="${searchName}"]`);
    }

    const count = await addToCartButton.count();
    if (count > 0) {
      await addToCartButton.first().click();
      return;
    }
  }

  /**
   * Apply filter
   * @param {string} filterType - Type of filter
   * @param {string} filterValue - Value to filter by
   */
  async applyFilter(filterType, filterValue) {
    try {
      // Try to find filter dropdown or checkbox
      const filterElement = this.page.locator(`[data-filter="${filterType}"], .filter-${filterType.toLowerCase()}`);

      if (await filterElement.isVisible()) {
        // If it's a dropdown
        if (await filterElement.locator('select').count() > 0) {
          await filterElement.locator('select').selectOption(filterValue);
        }
        // If it's a checkbox/radio
        else if (await filterElement.locator(`input[value="${filterValue}"]`).count() > 0) {
          await filterElement.locator(`input[value="${filterValue}"]`).check();
        }
        // If it's a clickable option
        else {
          const optionElement = filterElement.getByText(filterValue);
          if (await optionElement.isVisible()) {
            await optionElement.click();
          }
        }

        // Wait for filter to apply
        await this.page.waitForTimeout(1000);
        await this.waitForPageLoad();
      }
    } catch (error) {
      throw new Error(`Failed to apply filter ${filterType}=${filterValue}: ${error.message}`);
    }
  }

  /**
   * Sort products
   * @param {string} sortOption - Sort option (e.g., 'price', 'name', 'popularity')
   */
  async sortProducts(sortOption) {
    try {
      // Find sort dropdown or clickable element
      const sortElement = this.page.getByText('Sort', { exact: false }).first();
      
      if (await sortElement.isVisible()) {
        await sortElement.click();
        
        // Click on the specific sort option
        const option = this.page.getByText(sortOption, { exact: false });
        if (await option.isVisible()) {
          await option.click();
        }

        // Wait for sort to apply
        await this.page.waitForTimeout(1000);
        await this.waitForPageLoad();
      } else {
        throw new Error(`Could not find sort element`);
      }
    } catch (error) {
      throw new Error(`Failed to sort products by ${sortOption}: ${error.message}`);
    }
  }
}

export default ProductCatalogPage;
